### DISCUSSION TITLE
multiple same workflow transition at same time

### DISCUSSION THREAD

**Discussion by Floberrot** [symfony/symfony#50725] [Q&A] [answered]
Hi everybody ! 

With my team, we would like to know if there is a way to block multiple calls at the same time for a single transition.   
To explain, we have a VueJS  front that call an endpoint to launch a transition on a object, but, some users manage to double-click on it in spite of a security front. So, we would like to check in a Guard event or before if the transition is already launch and block the next one in this case.

Thanks a lot !

**Comment by yun548** [comment, marked as answer]
You could use the [Lock component](https://symfony.com/doc/current/components/lock.html) to prevent concurrent requests on a same object.

**Reply by Floberrot** [reply]
thanks a lot !

**Comment by ghost** [comment]
>  spite of a security front

How is a user able to double click if frontend secures it? Ofc I can imagine very very fast second click that happens between the first click and disabling button in vue but when the handler function is called at the second time it could check the current state which should be a disabled state after the first click and api call should not be executed.
